src/python-common/ceph/rgw/rgwam_core.py

```diff
@@ -386,101 +430,125 @@ def zone_op(self):
+            zone_info = self.zone_op().create(realm, zg,
+                                              EntityName(zone_name),
+                                              endpoints,
+                                              is_master=zone_is_master,
+                                              access_key=access_key,
+                                              secret=secret)
+
+            zone = EntityKey(zone_info['name'], zone_info['id'])
+            logging.info(f'Created zone name={zone.name} id={zone.id}')
+            return zone
         except RGWAMException as e:
             raise RGWAMException('failed to create zone', e)
 
-        zone_name = zone_info['name']
-        zone_id = zone_info['id']
-        logging.info('Created zone %s (%s)' % (zone_name, zone_id))
-
-        zone = EntityName(zone_name)
-
+    def create_system_user(self, realm, zonegroup, zone):
         try:
-            period_info = self.period_op().update(realm, EntityName(zg_name), zone, commit=True)
-        except RGWAMCmdRunException as e:
-            raise RGWAMException('failed to update period', e)
-
-        period = RGWPeriod(period_info)
-
-        logging.info('Period: ' + period.id)
-
-        try:
-            sys_user_info = self.user_op().create(zone, zg, uid=sys_uid, uid_prefix='user-sys',
+            sys_user_info = self.user_op().create(zone,
+                                                  zonegroup,
+                                                  uid=f'sysuser-{realm.name}',
+                                                  uid_prefix='user-sys',
                                                   is_system=True)
+            sys_user = RGWUser(sys_user_info)
+            logging.info('Created system user: %s' % sys_user.uid)
+            access_key = sys_user.keys[0].access_key if sys_user and sys_user.keys else ''
+            secret_key = sys_user.keys[0].secret_key if sys_user and sys_user.keys else ''
+            sys_user.add_key(access_key, secret_key)
+            return sys_user
         except RGWAMException as e:
             raise RGWAMException('failed to create system user', e)
 
-        sys_user = RGWUser(sys_user_info)
-
-        logging.info('Created system user: %s' % sys_user.uid)
-
-        sys_access_key = ''
-        sys_secret = ''
-
-        if len(sys_user.keys) > 0:
-            sys_access_key = sys_user.keys[0].access_key
-            sys_secret = sys_user.keys[0].secret_key
-
-        try:
-            zone_info = self.zone_op().modify(zone, zg, endpoints, None, sys_access_key, sys_secret)
-        except RGWAMException as e:
-            raise RGWAMException('failed to modify zone info', e)
-
+    def create_normal_user(self, zg, zone, uid=None):
         try:
             user_info = self.user_op().create(zone, zg, uid=uid, is_system=False)
+            user = RGWUser(user_info)
+            logging.info('Created regular user: %s' % user.uid)
```

Comment:
same NIT, can we include zonegoup and zone in this log?

Reply:
Fixed